src/error.rs

```diff
@@ -0,0 +1,33 @@
+use std::error::Error;
+use std::fmt;
+
+#[derive(Debug)]
+pub struct InfoError {
+    details: String,
+}
+
+impl InfoError {
+    pub fn new(msg: &str) -> InfoError {
```

Comment:
nit consider allowing anything that can be converted to a string
```suggestion
    pub fn new(msg: impl ToString) -> InfoError {
```